File: program/program/check-special-character/check_special_character.dart
void main() {
  final regExp = RegExp(
    r'[\^$*.\[\]{}()?\-"!@#%&/\,><:;_~`+='
    "'"
    ']'
  );
  
  final string = '@';

  if(string.contains(regExp))
    print("Special Character");
  else
    print("Not a Special Character");
}

File: program/program/check-special-character/CheckSpecialCharacter.java
import java.util.*;

public class SpecialChar {

    public static void main(String[] args) {

        String specialCharacters = " !#$%&'()*+,-./:;<=>?@[]^_`{|}";

        System.out.println("Enter the Character : ");
        Scanner scanner = new Scanner(System.in);

        char c = scanner.next().charAt(0);
        if (specialCharacters.contains(Character.toString(c))) {

            System.out.println("Special character");
        } else {
            System.out.println("Not a Special character");

        }

    }

}
